src/flow-manager.c

```diff
@@ -983,6 +984,8 @@ static TmEcode FlowRecycler(ThreadVars *th_v, void *thread_data)
 
                 (void)OutputFlowLog(th_v, ftd->output_thread_data, f);
 
+                if (f->flags & FLOW_HAS_EXPECTATION)
```

Comment:
Can you move this to `FlowClearMemory`, for two reasons:

- just cleaner
- this will cover a case that is currently missed as well: `FlowGetUsedFlow`